src/Core/Silk.NET.BuildTools/Converters/Readers/OpenCLReader.cs

```diff
@@ -835,42 +755,181 @@ public IEnumerable<Enum> ReadEnums(object obj, BindTask task)
 
             var registry = doc.Element("registry");
             Debug.Assert(registry != null, $"{nameof(registry)} != null");
-            
-            var allEnums = registry
+
+            var apis = registry.Elements("feature").Concat
+            (
+                registry.Elements("extensions").Elements("extension")
+                ?? throw new InvalidDataException()
+            );
+            Debug.Assert(apis != null, nameof(apis) + " != null");
+
+            var ulongEnums = new HashSet<string>(registry
+                .Elements("types")
+                .Elements("type")
+                .Where(e => e.Elements("type").SingleOrDefault()?.Value == "cl_bitfield")
+                .Select(e => Rename(e.Element("name").Value, task).Renamed));
+
+            var flagEnums = new HashSet<string>(registry
+                .Elements("enums")
+                .Where(e => e.Attribute("type")?.Value == "bitmask")
+                .Select(e => Rename(e.Attribute("name").Value, task).Renamed));
+
+            var enumEntries = new Dictionary<RenamedEntry, HashSet<RenamedEntry>>();
+            var enumTypes = new Dictionary<RenamedEntry, HashSet<RenamedEntry>>();
+
+            void AddEntry(RenamedEntry key, RenamedEntry value, Dictionary<RenamedEntry, HashSet<RenamedEntry>> dict)
+            {
+                if (dict.TryGetValue(key, out var list))
+                {
+                    list.Add(value);
+                }
+                else
+                {
+                    dict.Add(key, new HashSet<RenamedEntry> { value });
+                }
+            }
+
+            var constants = new HashSet<RenamedEntry>(registry
+                .Elements("enums")
+                .Where(x => x.Attribute("name").Value.StartsWith("Constants") || 
+                            x.Attribute("name").Value == "MiscNumbers")
+                .Elements("enum")
+                .Select(x => Rename(x.Attribute("name").Value, task))
+                .Concat(Constants.Keys.Select(c => Rename(c, task))));
+
+            var enumsPass1 = registry
                 .Elements("enums")
                 .Elements("enum")
-                .DistinctBy(x => x.Attribute("name")?.Value)
                 .Where
                 (
-                    x => x.Parent?.Attribute("name")?.Value != "Constants" &&
-                         x.Parent?.Attribute("name")?.Value != "MiscNumbers"
+                    x => !x.Parent.Attribute("name").Value.StartsWith("Constants") &&
+                         !x.Parent.Attribute("name").Value.StartsWith("enums") &&
+                         x.Parent.Attribute("name").Value != "MiscNumbers" &&
+                         x.Parent.Attribute("name").Value != "cl_device_info"
```

Comment:
Can you add a comment explaining why these can't make the first pass?

Reply:
Yeah, it's basically constants, plus unnamed, plus bug in XML with cl_device_info that makes it unusable